### PR TITLE
Doc/fix legal form specification generation

### PR DESCRIPTION
**Pull request by sauterl**
Fixes Company#LegalForm Enum values not correctly documented in generated openapi specification.

## Proposed Changes

 - Maps the enum values instead of the actual enum
 - Enum values are documented in schema desription
 - Workaround as this is not supported by swagger (?)
 - Open for discussion

## Definition of Done (DoD) - criteria met

 - [x] code compiles and [build is green](https://travis-ci.org/baloise/corellia)
 - [ ] SonarQube Rating A
 - [x] documentation is appropriate or has been updated accordingly
 - [x] [Codecov](https://codecov.io/gh/baloise/corellia) test execution coverage not lowered (aim: 80%)
 - [x] change is to be merged on master branch
 - [ ] change has been deployed in target environment for acceptance testing

### CODE REVIEW

**Comment by sauterl** [comment]
This is a workaround solution using [Swagger's Extension API](https://github.com/swagger-api/swagger-core/wiki/Swagger-2.X---Extensions).
It provides a custom converter for enums that use Jackson's @JsonFormat(shape=JsonFormat.Shape.OBJECT) and thus provide a getter annotated with @JsonValue
 
However, I'm not sure whether this is helpful for clients using the generated API specs. Hence, this PR is open for discussion

File: src/main/java/ch/baloise/corellia/api/doc/JsonFormatObjectEnumConverter.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                  try {
+                    values.put(e.name(), method.invoke(enumConstant));
+                  }
+                  catch (IllegalAccessException | InvocationTargetException ex) {
+                    ex.printStackTrace(); // Could be safely ignored, as the method shall exist and there is no security
+                  }
+                }
+                /*
+                // Alternative: create custom key-value pairs. However this is not very effektive.
+                ObjectSchema objectSchema = new ObjectSchema();
```
Comment (dweber019) [inline src/main/java/ch/baloise/corellia/api/doc/JsonFormatObjectEnumConverter.java]: Remove unused code.

File: src/main/java/ch/baloise/corellia/api/doc/JsonFormatObjectEnumConverter.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    }
+    if(chain.hasNext()){
+      return chain.next().resolve(type,context,chain);
+    }else{
+      return null;
+    }
+  }
+
+  private static boolean hasAnnotation(AccessibleObject object, Class<? extends Annotation> anCls){
+    for(Annotation an : object.getAnnotations()){
```
Comment (dweber019) [inline src/main/java/ch/baloise/corellia/api/doc/JsonFormatObjectEnumConverter.java]: You could use `anyMatch` here.

File: src/main/java/ch/baloise/corellia/api/doc/JsonFormatObjectEnumConverter.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+/**
+ * A {@link ModelConverter} dedicated for enum using {@link JsonValue} annoation on a getter.
+ * This is then used to create a specification for a enum in the form of ENUM_INSTANCE_NAME: VALUE.
+ */
+public class JsonFormatObjectEnumConverter implements ModelConverter {
+
+  private JsonPropertyDescriptionResolver jsonPropertyDescriptionResolver = new JsonPropertyDescriptionResolver(Json.mapper());
+
+  @Override
+  public Schema resolve(AnnotatedType type, ModelConverterContext context, Iterator<ModelConverter> chain) {
```
Comment (dweber019) [inline src/main/java/ch/baloise/corellia/api/doc/JsonFormatObjectEnumConverter.java]: This method does a lot, could we refactor or extract stuff into separate methods?

File: src/main/java/ch/baloise/corellia/api/doc/JsonFormatObjectEnumConverter.java (hunk 1 above)
Comment (sauterl) [reply src/main/java/ch/baloise/corellia/api/doc/JsonFormatObjectEnumConverter.java]: Done, see 498ac2b

File: src/main/java/ch/baloise/corellia/api/doc/JsonFormatObjectEnumConverter.java (hunk 2 above)
Comment (sauterl) [reply src/main/java/ch/baloise/corellia/api/doc/JsonFormatObjectEnumConverter.java]: Done, see 498ac2b

File: src/main/java/ch/baloise/corellia/api/doc/JsonFormatObjectEnumConverter.java (hunk 3 above)
Comment (sauterl) [reply src/main/java/ch/baloise/corellia/api/doc/JsonFormatObjectEnumConverter.java]: Done, see 498ac2b

**Comment by sauterl** [comment]
As we internally discussed, we will not use this feature.

**Pull request closed without merging**